src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -183,8 +208,11 @@ public void UpdatePackageVersion(PackageInfo packageInfo)
                     (string? original, XDocument? document) pair = (null, null);
                     if (PackagePropFileToDocumentMap.TryGetValue(file, out pair!))
                     {
-                        var propsVersionAttribute = GetVersionAttributeInPropsFileOpt(pair.document!, packageInfo);
-                        propsVersionAttribute?.SetValue(packageInfo.Version.ToString());
+                        var versionObject = GetVersionAttributeOrElementInPropsFile(pair.document!, packageInfo);
+                        if (versionObject is XAttribute attribute)
+                            attribute.SetValue(packageInfo.Version.ToString());
+                        else if (versionObject is XElement element)
+                            element.SetValue(packageInfo.Version.ToString());
```

Comment:
should we do something (throw / log) if neither is true?

Reply:
Ah, probably. I don't understand why the null ?. was there before, but you're right this makes it worse if the function were to return non-null but some other type.

Reply:
Added a throw.